vmm/src/device_manager.rs

```diff
@@ -1180,6 +1183,12 @@ impl DeviceManager {
             console_resize_pipe,
         )?;
 
+        let tpm = self.config.lock().unwrap().tpm.clone();
+        if tpm.is_some() {
+            let vtpm = self.add_tpm_device(&legacy_interrupt_manager, tpm.unwrap().path)?;
+            self.bus_devices
+                .push(Arc::clone(&vtpm) as Arc<Mutex<dyn BusDevice>>);
+        }
```

Comment:
Please replace with the following syntax:
```suggestion
        if let Some(tpm) = self.config.lock().unwrap().tpm.as_ref() {
            let tpm_dev = self.add_tpm_device(&legacy_interrupt_manager, tpm.path)?;
            self.bus_devices
                .push(Arc::clone(&tpm_dev) as Arc<Mutex<dyn BusDevice>>);
        }
```